packages/odyssey-react/src/components/Modal/Modal.tsx

```diff
@@ -111,9 +122,16 @@ export const Modal = withTheme(
     const oid = useOid(id);
     const modalDialog = useRef<HTMLDivElement>(null);
     const componentClass = useCx(styles.root, { [styles.openState]: open });
-
-    if (open && onOpen) {
-      onOpen();
+    const { restoreFocus, setFocus } = useFocus();
+    const lastFocusedElemRef = useRef<OptionalHTMLElement>(null);
+
+    if (open) {
+      lastFocusedElemRef.current = setFocus(modalDialog.current);
+      if (onOpen) {
```

Comment:
minor, but can we avoid the nested conditional here? They tend to become hard to reason about. Even something simple like this would work:

```
if (open) {
  lastFocusedElemRef.current = setFocus(modalDialog.current);
  onOpen && onOpen();
} else {
  lastFocusedElemRef.current && restoreFocus(lastFocusedElemRef.current);
}